jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2368,21 +2377,107 @@ private ClassDefiner classDefiner(byte[] bytes, ClassOption... classOptions) thr
 				throw new IllegalAccessException();
 			}
 
+			int flags = 0;
+			for (ClassOption opt : classOptions) {
+				flags |= opt.toFlag();
+			}
+			String targetClassName = getClassNameAndCheckClassBytes(bytes);
+			return makeHiddenClassDefiner(targetClassName, bytes, flags);
```

Comment:
> Why does this need the name here?

Because the new API `JavaLangAccess.defineClass()` added by openjdk to define hidden class takes className as a parameter, so we pass className to makeHiddenClassDefiner() which eventually calls JavaLangAccess.defineClass(). 

The checks for magic number and class version are moved into cfreader.c.